cmd/nodepool/azure/create.go

```diff
@@ -54,6 +67,41 @@ func (o *AzurePlatformCreateOptions) UpdateNodePool(_ context.Context, nodePool
 		return fmt.Errorf("resource-group-name is required when using disk-encryption-set-id")
 	}
 
+	var vmImage *hyperv1.AzureVMImage
+	if o.ImageID != "" {
+		vmImage = &hyperv1.AzureVMImage{
+			Type: hyperv1.ImageID,
+			ImageID: &hyperv1.AzureImageInfo{
+				ID: ptr.To(o.ImageID),
+			},
+		}
+	} else {
+		// Validate publisher, offer, sku, and version flags are provided if one of them is not empty
```

Comment:
I'm missing where this is being enforced? same as https://github.com/openshift/hypershift/pull/4191/commits/55b2a197f07fe55f04f8df184288ef8f91e42f66#r1655066371

Reply:
That's taken care of here - https://github.com/openshift/hypershift/blob/82c92662715c65de4e9b6c855c71e23a68ac88a2/cmd/nodepool/azure/create.go#L83